pkg/operator/discover/discover.go

```diff
@@ -81,10 +81,7 @@ func (d *Discover) Start(ctx context.Context, namespace, discoverImage, security
 }
 
 func (d *Discover) createDiscoverDaemonSet(ctx context.Context, namespace, discoverImage, securityAccount string, data map[string]string, useCephVolume bool) error {
-	discoveryInterval := k8sutil.GetValue(data, discoverIntervalEnv, "")
-	if discoverImage == "" {
-		discoveryInterval = defaultDiscoverInterval
-	}
+	discoveryInterval := k8sutil.GetValue(data, discoverIntervalEnv, defaultDiscoverInterval)
```

Comment:
this is better than what it is currently, but it should work as expected as default values are being set.
what is the issue you are facing?

Reply:
I'm facing the error in the description, all discover pods crash on startup as the value passed is `""`
I used the Helm chart for the operator with most of the default values and when I enabled the discovery Daemon all pod crash looped.

From what I gathered the if condition above is not checking the right variable to decide to set `discoveryInterval` to `defaultDiscoverInterval`

Looking into what source `discoverImage` value it seems to come from `ReconcileConfig.config.Image` which in turn seems to be set by default in the helm chart: https://github.com/rook/rook/blob/master/deploy/charts/rook-ceph/values.yaml#L5 . So in my case that if would never evaluate to true.

Reply:
> can you share operator logs where we logs all the value modifed/added

[operator.log.txt](https://github.com/rook/rook/files/12512933/operator.log.txt)

```
2023-09-04 10:45:55.898928 I | op-k8sutil: ROOK_DISCOVER_DEVICES_INTERVAL="" (default)
```

Reply:
While I agree the change is better style, I don't see how it changes the behavior. This check in the previous implementation should ensure the interval is not empty:
```
	if discoverImage == "" {
		discoveryInterval = defaultDiscoverInterval
	}
```
So I don't see what is causing your failure.

Reply:
Unless I missed another place where `discoveryInterval` is set, the previous implementation doesn't set the default value if `k8sutil.GetValue` doesn't find  `discoverIntervalEnv` 

To work as intended it would need to be changed from
```diff
  discoveryInterval := k8sutil.GetValue(data, discoverIntervalEnv, "")
- if discoverImage == "" {
		discoveryInterval = defaultDiscoverInterval
	}
```
To
``` diff
  discoveryInterval := k8sutil.GetValue(data, discoverIntervalEnv, "")
+ if discoveryInterval == "" {
		discoveryInterval = defaultDiscoverInterval
	}
```

But I propose to set the default value directly in the call to `k8sutil.GetValue` instead

Reply:
Oh right, that was the bug. `discoverImage` was the wrong variable, and caused this issue. Now it makes sense that this fixes the issue.